backend/mirai-console/build.gradle.kts

```
@@ -76,4 +77,79 @@ tasks {
     }
 }
 
-configurePublishing("mirai-console")
\ No newline at end of file
+configurePublishing("mirai-console")
+
+setupTesterPluginSources()
+
+fun Project.setupTesterPluginSources() {
+    val testers = 2
+    val taskNames = Array(testers + 1) { index ->
+        if (index == testers) {
+            "jarAutoEnd"
+        } else {
+            "jarTesterPlugin$index"
+        }
+    }
+    sourceSets {
+        val baseDir = project.projectDir.resolve("tester-plugins")
+        fun newTester(name: String, index: Int) = create(name) {
+            val consoleMain = this@sourceSets.getByName("main")
+            val consoleTest = this@sourceSets.getByName("test")
+            // Only need compile classpath
+            compileClasspath += consoleMain.output
+            compileClasspath += consoleMain.compileClasspath
+
+            compileClasspath += consoleTest.output
+            compileClasspath += consoleTest.compileClasspath
+
+            resources.setSrcDirs(listOf(baseDir.resolve(name)))
+            java.setSrcDirs(listOf(baseDir.resolve(name)))
+
+        }.also { src ->
+
+            project.kotlin.target.compilations.run {
+                maybeCreate(src.name).let { ksrc ->
+                    ksrc.associateWith(getByName("main"))
```

```suggestion
                    ksrc.associateWith(getByName("main"))
```

Kotlin stdlib 的 `associateWith`? 那它是 redundant 的